lib/bundler/inject.rb

```diff
@@ -1,6 +1,42 @@
 require "bundler/inject/version"
 require "bundler/inject/dsl_patch"
 
+module Bundler
+  module Inject
+    class << self
+      # Check if we should skip outputing warnings
+      #
+      # This can be set in two ways:
+      #
+      #   - Via bundler's Bundler::Settings
+      #   - When RAILS_ENV=production is set
+      #
+      # To configure the setting, you can run:
+      #
+      #   bundle config bundler_inject.disable_warn_override_gem true
+      #
+      # OR use an environment variable
+      #
+      #   BUNDLE_BUNDLER_INJECT__DISABLE_WARN_OVERRIDE_GEM=true bundle ...
+      #
+      # If neither are set, it will check for ENV["RAILS_ENV"] is "production",
+      # and will skip if it is, but the bundler variable is present (and even
+      # set to "false") that will be favored.
+      def skip_warnings?
+        return @skip_warnings if defined?(@skip_warnings)
+
+        bundler_setting = Bundler.settings["bundler_inject.disable_warn_override_gem"]
```

Comment:
Does Bundler.settings convert the setting to a boolean?   I'm only asking because if someone sets `BUNDLE_BUNDLER_INJECT__DISABLE_WARN_OVERRIDE_GEM=false` as you did in the test, wouldn't it return the String `"false"` which is truthy?

Reply:
It does and it doesn't.

When encounters a "false" string:

https://github.com/bundler/bundler/blob/264ef6be/lib/bundler/settings.rb#L387

It will convert it to `false`, otherwise it will basically return it's value (so "truthy").

* * *

The other thing of note is how it returns a missing key, which can be found here:

https://github.com/bundler/bundler/blob/264ef6be/lib/bundler/settings.rb#L94-L105

It will check all the configs in order, and if none of them match a key, `nil` is returned. So when coding this, I explicitly made it so that if `bundler_setting.nil?`, use the fallback since that is when a key doesn't exist.

Reply:
Note:  While there is logic in `Bundler::Settings` to [convert "false-like" values to boolean](https://github.com/bundler/bundler/blob/264ef6be/lib/bundler/settings.rb#L333-L340), this only happens for keys that are known to `Bundler` proper.

The `elsif is_bool(key) || value == "false"` is what allows the above to work for "false" values that aren't known to `bundler`, but we don't get the added benefit of being able to use other values like "f", "no", "0", ect.

(I incorrectly thought you could use those extra values originally in the previous post, but caught this prior to posting)

Reply:
Also also... 

> wouldn't it return the String `"false"` which is truthy?

I did write a test around this for specifically this reason 😉